benchmarks/run_benchmark.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import torchvision
+import torch
+try:
+    import transformers
+except:
+    pass
+from torchvision.prototype.datasets import load
+import torch.nn.functional as F
+from torchvision import transforms
+import time
+from statistics import mean
+import torch.optim as optim
+
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--dataset", type=str, default="gtsrb", help="The name of the dataset")
+parser.add_argument("--model_name", type=str, default="resnext50_32x4d", help="The name of the model")
+parser.add_argument("--batch_size", type=int, default=1, help="")
+parser.add_argument("--device", type=str, default="cuda:0", help="Options are are cpu or cuda:0")
+parser.add_argument("--num_epochs", type=int, default=2)
+parser.add_argument("--report_location", type=str, default="./report.md", help="The location where the generated report will be stored")
+parser.add_argument("--num_workers", type=int, default=1, help="Number of dataloader workers")
+parser.add_argument("--shuffle", action="store_true")
+parser.add_argument("--dataloaderv", type=int, default=1)
+
+args = parser.parse_args()
+dataset = args.dataset
+model_name = args.model_name
+batch_size = args.batch_size
+num_epochs = args.num_epochs
+report_location = args.report_location
+num_workers = args.num_workers
+shuffle = args.shuffle
+dataloaderv = args.dataloaderv
+
+if dataloaderv == 1:
+    from torch.utils.data import DataLoader
+elif dataloaderv == 2:
+    from torch.utils.data.dataloader_experimental import DataLoader2 as DataLoader
+else:
+    raise(f"dataloaderv{dataloaderv} is not a valid option")
+
+# Util function for multiprocessing
+def init_fn(worker_id):
+    info = torch.utils.data.get_worker_info()
+    num_workers = info.num_workers
+    datapipe = info.dataset
+    torch.utils.data.graph_settings.apply_sharding(datapipe, num_workers, worker_id)
```

Comment:
We should make sure we can remove this eventually as users shouldn't have to shard manually with DataLoaderV2